src/rgw/driver/rados/rgw_sal_rados.h

```diff
@@ -948,6 +949,82 @@ class RadosLifecycle : public Lifecycle {
 						       const std::string& cookie) override;
 };
 
+class RestoreRadosSerializer : public StoreRestoreSerializer {
+  librados::IoCtx& ioctx;
+  ::rados::cls::lock::Lock lock;
+
+public:
+  RestoreRadosSerializer(RadosStore* store, const std::string& oid, const std::string& lock_name, const std::string& cookie);
+
+  virtual int try_lock(const DoutPrefixProvider *dpp, utime_t dur, optional_yield y) override;
+  virtual int unlock() override {
+    return lock.unlock(&ioctx, oid);
+  }
+};
+
+class RadosRestore : public Restore {
+  RadosStore* store;
+  std::string prefix;
+  int num_objs;
+  librados::IoCtx& ioctx;
+  using centries = std::vector<ceph::buffer::list>;
+  ceph::containers::tiny_vector<LazyFIFO> fifos;
```

Comment:
@adamemerson I used `LazyFIFO` (from `rgw_log_backing.h`) to store restore entries similar to datalog. After rebase I noticed that you updated `LazyFIFO` to use `neorados`.  Do you suggest I update this `RadosRestore` too with neorados or do I need to create new wrapper for FIFO class ?